### PR TITLE
COMMERCE-11179 Generic Payment Gateway Engine

### PR DESCRIPTION
**Pull request by luca-pellizzon**
[COMMERCE-11179](https://issues.liferay.com/browse/COMMERCE-11179)

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:stop

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:stop

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/gateway/CommercePaymentGatewayImpl.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+import com.liferay.portal.kernel.util.HashMapBuilder;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	configurationPid = "com.liferay.commerce.payment.configuration.CommercePaymentEntryAuditConfiguration",
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/gateway/CommercePaymentGatewayImpl.java:42]: check if you still need this

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
@luca-pellizzon I did a double-check on the relevant failures, it doesn't look like there are failure related to the PR. This can be moved forward.

**Comment by luca-pellizzon** [comment]
ci:forward:force

**Pull request closed without merging**